state_machine/applications/system-check/tests/coil_test.py

```diff
@@ -0,0 +1,130 @@
+"""
+Python system check script for PyCubed satellite board
+PyCubed Mini mainboard-v02 for Pocketqube Mission
+Torque Driver Test
+* Author(s): Yashika Batra
+"""
+
+import time
+from lib import pycubed as cubesat
+from math import sqrt
+
+# voltage level constants; set between -1 and 1
+v1 = 0.25
+v2 = 0.5
+v3 = 0.75
+
+# find projected voltage: vlevel * max voltage
+projected_voltage1 = v1 * 5.06
+projected_voltage2 = v2 * 5.06
+projected_voltage3 = v3 * 5.06
+
+
+def user_test(coil_index):
+    """
+    All user interaction happens in this function
+    Set wait times, prompt user, collect magnetometer data for
+    different voltage levels
+    """
+
+    # wait times
+    wait_time = 3
+    driver_time = 3
+
+    # set up the test
+    print("Testing Coil", str(coil_index),
+          "for the following voltage levels: " + str(projected_voltage1) +
+          ", " + str(projected_voltage2) + ", " + str(projected_voltage3))
+    print("Waiting", str(wait_time), "seconds.")
+    time.sleep(wait_time)
+
+    # conduct the test
+    print("Beginning the test now. Testing for", str(driver_time * 3),
+          "seconds.")
+
+    # test each voltage level
+    cubesat.coildriver_vout(coil_index, projected_voltage1)
+    time.sleep(driver_time)
+    mag_reading1 = cubesat.magnetic()
+
+    cubesat.coildriver_vout(coil_index, projected_voltage2)
+    time.sleep(driver_time)
+    mag_reading2 = cubesat.magnetic()
+
+    cubesat.coildriver_vout(coil_index, projected_voltage3)
+    time.sleep(driver_time)
+    mag_reading3 = cubesat.magnetic()
+
+    # calculate total magnetic reading
+    print("Data Collection Complete")
+    mag_total1 = sqrt(
+        mag_reading1[0] ** 2 + mag_reading1[1] ** 2 + mag_reading1[2] ** 2)
+    mag_total2 = sqrt(
+        mag_reading2[0] ** 2 + mag_reading2[1] ** 2 + mag_reading2[2] ** 2)
+    mag_total3 = sqrt(
+        mag_reading3[0] ** 2 + mag_reading3[1] ** 2 + mag_reading3[2] ** 2)
+
+    # if magnetometer readings are increasing over time, return true
+    return (mag_total3 > mag_total2 and mag_total2 > mag_total1)
```

Comment:
I'd like some more detailed logging here if the tests fails.

Reply:
What exactly are your thoughts? I can print out the magnetometer readings in the case of failure

Reply:
Yup, that'd be perfect. 
Something like here were the 3 results, they should have been increasing but they weren't would be perfect.